documentation/Sin Cos.tex

```diff
@@ -390,9 +396,172 @@ \subsection*{Fallback}
+
+\begin{algorithm}
+\DontPrintSemicolon
+\SetAlgoLined
+\LinesNumbered
+\SetNlSty{texttt}{}{.}
+\SetAlgoNlRelativeSize{0}
+\KwIn{The interval to search $\intclos{x_{\text{mid}} - \frac{T}{N}}{x_{\text{mid}} + \frac{T}{N}}$ defined by its midpoint $x_{\text{mid}}$ and its scaled radius $T$,
+and a positive integer $M$.}
+\KwOut{A $t \in \intclos{-T}{T}$ such that $x_{\text{mid}} + \frac{t}{N}$ is an accurate value for $\sin$ and $\cos$; furthermore, if the search direction is $\texttt{UP}$, $t$ is as close as possible to $-T$ and the search direction is $\texttt{DOWN}$, $t$ is as close as possible to $T$.}
+\SetKwFunction{SimultaneousBadCaseSearch}{SimultaneousBadCaseSearch}
+Let $F_1 = \FunctionBody{t}{N \sin\of{x_{\text{mid}} + \frac{t}{N}}}$ and $F_2 = \FunctionBody{t}{N \cos\of{x_{\text{mid}} + \frac{t}{N}}}.$\;
+\uIf{$T$ < some threshold}{
+  \Switch{search direction}{
+    \uCase{$\texttt{UP}$}{
+    	  \ForEach{$t_0 \in \intclos{-T}{T}$ by ascending values}{
+        \If{$\abs{F_i\of{t_0} \cmod 1} < \frac{1}{M}$ for all $i \in \set{1, 2}$}{
+          \Return{$t_0$.}
+        }
+      }
+    }
+    \uCase{$\texttt{DOWN}$}{
+    	  \ForEach{$t_0 \in \intclos{-T}{T}$ by descending values}{
+        \If{$\abs{F_i\of{t_0} \cmod 1} < \frac{1}{M}$ for all $i \in \set{1, 2}$}{
+          \Return{$t_0$.}
+        }
+      }
+    }
+  }
+}
+\Else{
+  \Return \SimultaneousBadCaseSearch{$F_1$, $F_2$, $M$, $T$}.\;
+}
+\caption{IntervalSimultaneousBadCaseSearch\label{ISBCS}.}
+\end{algorithm}
+
+Algorithm \ref{ISBCS} must be integrated into a retry loop that searches progressively farther above and below $2 k \gD$ until a solution is found.  This is what algorithm \ref{FSBCS} is doing: it first searches an interval of radius $T_0$ above $2 k \gD$, and retries with intervals progressively smaller if algorithm \ref{ISBCS} returns a failure.  When algorithm \ref{ISBCS} returns an empty solution set, the last interval searched is marked as ``covered'' and the algorithm proceeds to perform a similar search below $2 k \gD$.  If that search does not find a solution, a new interval of radius $T_0$ above the covered interval is tried, and so on repeatedly.  The algorithm returns when a solution is found.  That solution is nearly optimal in terms of its distance to $2 k \gD$ with the caveat that, since we search above $2 k \gD$ before searching below, we could return a solution above when there is a slightly closer solution below.
+
+The value of $T_0$ is chosen based on the Coppersmith bound established in \cite{StehléZimmermann2005}: $T^3 = \BigO\of{M · N}$.  Evidently this bound does not unambiguously determine $T_0$, but in practice $T_0 = \sqrt[3]{M N}$ yields good results: a value too large would cause too many splits of the search interval before finding a solution or (more likely) rejecting the interval; a value too small would require to process too many small intervals. 
```

Comment:
```suggestion
The value of $T_0$ is chosen based on the Coppersmith bound established in \cite{StehléZimmermann2005}: $T^3 = \BigO\of{M \cdot N}$.  Evidently this bound does not unambiguously determine $T_0$, but in practice $T_0 = \sqrt[3]{M N}$ yields good results: a value too large would cause too many splits of the search interval before finding a solution or (more likely) rejecting the interval; a value too small would require to process too many small intervals. 
```

We want U+22C5, not U+00B7.

Reply:
There are more occurrences, doing them all.